hxd/res/BDFFont.hx

```diff
@@ -0,0 +1,415 @@
+		var line : String;
+		var prop : String;
+		var args : Array<String>;
+
+		// Iterate lines
+		while (lines.length>0) {
+			linenum++;
+
+			line = lines.shift();
+			args = StringTools.trim(line).split(" ");
+			if (args.length==0) continue;
+			prop = StringTools.trim(args.shift());
+
+			switch (prop) {
+				case 'FAMILY_NAME':
+					font.name = extractStr(args);
+				case 'SIZE':
+					font.size = font.initSize = extractInt(args[0]);
+				case 'BITS_PER_PIXEL':
+					this.bitsPerPixel = extractInt(args[0]);
+					if (!Lambda.has([1,2,4,8],bitsPerPixel)) throw('BITS_PER_PIXEL of '+bitsPerPixel+' not supported, at line '+linenum);
+				case 'FONTBOUNDINGBOX':
+					this.fbbHeight = extractInt(args[1]);
+				case 'FONT_ASCENT':
+					this.ascent = extractInt(args[0]);
+				case 'FONT_DESCENT':
+					this.descent = extractInt(args[0]);
+				// Once we find STARTCHAR we know that the header is done. Stop
+				// processing lines and continue.
+				case 'STARTCHAR':
+					break;	
+			}
+			
+
+		}
+		// Check we have everything we need
+		if (font.initSize==0) throw('SIZE not found or is 0');
+
+		// Return linenum we are up to
+		return linenum;
+	}
+
+	/**
+	 * Extract glyph information from the file.
+	 * @param lines     The remaining lines in the file
+	 * @param linenum   The current line number
+	 * @return Int      The final line number after processing header
+	 */
+	function parseGlyphs( lines : Array<String>, linenum : Int ) : Int{
+		var line : String;
+		var prop : String;
+		var args : Array<String>;
+
+		this.glyphData = new Array(); // Destroyed after generating bitmap
+
+		var processingGlyphHeader : Bool = true;
+		var encoding : Int = -1;
+		var stride : Int = -1;
+		var bbxFound : Bool = false;
+		var bbxWidth : Int = 0;
+		var bbxHeight : Int = 0;
+		var bbxXOffset : Int = 0;
+		var bbxYOffset : Int = 0;
+		var expectedLines : Int = 0;
+		var expectedBytesPerLine : Int = 0;
+		var char : BDFFontChar = null;
+
+		// Iterate lines
+		while (lines.length>0) {
+			linenum++;
+
+			line = lines.shift();
+			args = StringTools.trim(line).split(" ");
+			if (args.length==0) continue;
+			prop = StringTools.trim(args.shift());
+
+			// Start by processing the glyph header
+			if (processingGlyphHeader) {
+				switch (prop) {
+					case 'ENCODING':
+						// XXX: Support encoding ranges. Not sure if they are common? Hacen't come across this yet
+						if (encoding!=-1) throw ('Encoding ranges not supported, at line '+linenum);
+						encoding = extractInt(args[0]);
+						if (encoding<1) throw('ENCODING '+encoding+' not supported, at line '+linenum);
+					case 'DWIDTH': // Device width
+						stride = extractInt(args[0]);
+						if (stride<0) throw('DWIDTH is negative, at line '+linenum);
+						// XXX: This could be a warning and we could ignore the vertical step. Font might render weird?
+						if (extractInt(args[1])!=0) throw('A non-0 DWIDTH is not supported (maybe vertical character set?), at line '+linenum);
```

Comment:
```suggestion
						if ( encoding != -1 ) throw 'Encoding ranges not supported, at line $linenum';
						encoding = extractInt(args[0]);
						if ( encoding < 1 ) throw 'ENCODING $encoding not supported, at line $linenum';
					case 'DWIDTH': // Device width
						stride = extractInt(args[0]);
						if ( stride < 0 ) throw 'DWIDTH is negative, at line $linenum';
						// XXX: This could be a warning and we could ignore the vertical step. Font might render weird?
						if ( extractInt(args[1]) != 0 ) throw 'A non-0 DWIDTH is not supported (maybe vertical character set?), at line $linenum';
```